src/box/iproto_constants.h

```diff
@@ -218,7 +218,23 @@ extern const char *iproto_flag_bit_strs[];
 	/**
 	 * Flag indicating whether the transaction is synchronous.
 	 */								\
-	 _(IS_SYNC, 0x61, MP_BOOL)
+	 _(IS_SYNC, 0x61, MP_BOOL)					\
+	/**
+	 * Cursor, used in IRPOTO_FETCH_SNAPSHOT, in order to fetch snapshot
+	 * from a file:
+	 *
+	 * IPROTO_CURSOR: {
+	 *      IRPOTO_ENGINE: number,
```

Comment:
Why do you need the engine? I don't think that it's that bad to re-scan memtx snapshot if the replica failed to read vinyl, but even this could be avoided by remembering the number of rows in the memtx snapshot, without requiring the caller to specify the engine.

Reply:
I dropped engine, thank you for an idea!!! From now on we rewrite LSNs of the vinyl engine to create an ascending sequence of LSNs in order to make cleint think, that this is just one big snap file.